.github/workflows/test.yml

```diff
@@ -149,29 +149,35 @@ jobs:
           use-mamba: true
           python-version: "3.8"
           channel-priority: strict
+          channels: ${{ needs.detect-ci-trigger.outputs.triggered == 'true' && 'dask/label/dev,conda-forge,nodefaults' || 'conda-forge,nodefaults' }}
           activate-environment: dask-sql
           environment-file: continuous_integration/environment-3.8-jdk11-dev.yaml
       - name: Download the pre-build jar
         uses: actions/download-artifact@v1
         with:
           name: jar
           path: dask_sql/jar/
-      - name: Install dependencies
+      - name: Install cluster dependencies
         run: |
           mamba install python-blosc lz4 -c conda-forge
 
           which python
           pip list
           mamba list
-      - name: Optionally install upstream dev Dask / dask-ml
+      - name: Optionally install upstream dev dask-ml
         if: needs.detect-ci-trigger.outputs.triggered == 'true'
         run: |
-          python -m pip install --no-deps git+https://github.com/dask/dask
-          python -m pip install --no-deps git+https://github.com/dask/distributed
+          mamba update dask
```

Comment:
Despite including `dask/label/dev` in the `channels` of the Miniconda setup, it seems like the action prioritizes the channels of the environment file when creating an environment, meaning we only consider `conda-forge` packages:

https://github.com/dask-contrib/dask-sql/blob/cb3d9037b9c715e66238f27c68944837813ef7bb/continuous_integration/environment-3.10-jdk11-dev.yaml#L2-L4

The channel still has the highest priority after creating the environment though, so a `mamba update` ends up picking up the nightlies - overall this is pretty similar to what we were doing before by explicitly installing the packages with `mamba install -c dask/label/dev dask`, but allows us to keep the strict channel priority

Reply:
There's probably some hacky way to insert the `dask/label/dev` channel directly into the Conda environment file before creating the environment, which should allow us to remove this step altogether (if/once dask-ml nightlies are published); I can see if there's a trivial way to do that, otherwise I think this solution seems suitable

Reply:
Agreed this seems fine for now. More was trying to get a handle on what was going on. Could file an issue to follow up later